File: kombinasyon.cs
using System;

namespace kombinasyon
{
    class Program
    {
        static void Main(string[] args)
        {
            int sayiBir, sayiIki, sonuc;
            do
            {
                Console.Write("Lutfen birinci sayiyi giriniz: ");
                sayiBir = Convert.ToInt32(Console.ReadLine());
                Console.Write("Lutfen ikinci sayiyi giriniz: ");
                sayiIki = Convert.ToInt32(Console.ReadLine());
            }
            while (sayiIki > sayiBir);
            sonuc = fakt(sayiBir) / (fakt(sayiIki) * fakt(sayiBir - sayiIki)); //Hesaplama
            Console.WriteLine("C({0}, {1}) = {2}", sayiBir, sayiIki, sonuc);
            Console.ReadKey();
        }
        static int fakt(int gelen) //Ozyineli faktoriyel fonksiyonu
        {
            if (gelen == 0 || gelen == 1)
            {
                return 1;
            }
            else
            {
                return gelen * fakt(gelen - 1);
            }
        }
    }
}

File: Kombinasyon.swift
//  Created by Kağan on 15.01.2017.
//  Copyright © 2017 Kağan Utku Kılıçlı. All rights reserved.

import Foundation

func faktoriyel(_ a: Int) -> Int {
    var sonuc = 1
    for i in 1...a{
        sonuc = sonuc * i
    }
    return sonuc
}


func kombinasyon(_ n: Int,_ r: Int) -> Int {
    return faktoriyel(n) / (faktoriyel(r)*faktoriyel(n-r))
}

print("birinci sayiyi giriniz:")
if let sayiA = Int(readLine(strippingNewline: true)!) {
    print("ikinci sayiyi giriniz:")
    if let sayiB = Int(readLine(strippingNewline: true)!) {
        print("kombinasyon: \(kombinasyon(sayiA,sayiB))")
    }
}


